src/transaction/log_replication_atomic.cpp

```diff
@@ -383,9 +385,17 @@ namespace cublog
   {
     assert (m_locked_classes.count (trid) > 0);
 
-    for (auto classoid : m_locked_classes[trid])
+    if (m_locked_classes.size() == 1)
```

Comment:
Can you please explain this?
If this check does not take `trid` into account, how is it ensure that the single entry in the `m_locked_classes` is from this very `trid`? And we not are mistakenly unlocking resources from another ATS transaction executing concurrently with this `trid`.

Why not a more conservative approach: if we only want to unlock the `root_class_hold` just extract a function that does that out of `lock_unlock_all` and call that; this would also ensure that other possible mishandlings of lock manager are caught by lock manager verifications.

And if @wrlawodms comment below is valid, then only call this function (that unlocks the root class) in debug mode and leave it out for release mode.